src/main/java/io/supertokens/webserver/api/oauth/OAuthClientsAPI.java

```diff
@@ -0,0 +1,168 @@
+
+    private static final List<String> ALLOWED_INPUT_FIELDS = Arrays.asList(new String[]{"clientName","scope", "redirectUris", "allowedCorsOrigins", "authorizationCodeGrantAccessTokenLifespan", "authorizationCodeGrantIdTokenLifespan", "authorizationCodeGrantRefreshTokenLifespan",
+        "clientCredentialsGrantAccessTokenLifespan","implicitGrantAccessTokenLifespan","implicitGrantIdTokenLifespan","refreshTokenGrantAccessTokenLifespan","refreshTokenGrantIdTokenLifespan","refreshTokenGrantRefreshTokenLifespan","tokenEndpointAuthMethod","audience",
+        "grantTypes","responseTypes","clientUri","logoUri","policyUri","tosUri","metadata"});
+    private static final List<String> REQUIRED_INPUT_FIELDS = Arrays.asList(new String[]{"clientName", "scope"});
+
+    @Override
+    public String getPath() {
+        return "/recipe/oauth/clients";
+    }
+    public OAuthClientsAPI(Main main){
+        super(main, RECIPE_ID.OAUTH.toString());
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
+        JsonObject input = InputParser.parseJsonObjectOrThrowError(req);
+
+        boolean containsAllRequired = containsAllRequiredFields(input);
+        boolean containsMoreThanAllowed = containsMoreThanAllowed(input);
+        if(!containsAllRequired || containsMoreThanAllowed){
+            throw new ServletException(new WebserverAPI.BadRequestException("Invalid Json Input"));
+        }
+
+        try {
+            AppIdentifier appIdentifier = getAppIdentifier(req);
+            Storage storage = enforcePublicTenantAndGetPublicTenantStorage(req);
+
+            JsonObject response = OAuth.registerOAuthClient(super.main, appIdentifier, storage, input);
+            sendJsonResponse(200, response, resp);
+
+        } catch (OAuthClientRegisterException registerException) {
+
+            JsonObject errorResponse = createJsonFromException(registerException);
+            sendJsonResponse(400, errorResponse, resp);
+
+        } catch (TenantOrAppNotFoundException | InvalidConfigException | BadPermissionException
+                 | NoSuchAlgorithmException | StorageQueryException e) {
+            throw new ServletException(e);
+        }
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
+        String clientId = InputParser.getQueryParamOrThrowError(req, "clientId", false);
+
+        try {
+            AppIdentifier appIdentifier = getAppIdentifier(req);
+            Storage storage = enforcePublicTenantAndGetPublicTenantStorage(req);
+
+            JsonObject response = OAuth.loadOAuthClient(main, appIdentifier, storage, clientId);
+            sendJsonResponse(200, response, resp);
+
+        }  catch (OAuthClientException e) {
+            JsonObject errorResponse = createJsonFromException(e);
+            sendJsonResponse(400, errorResponse, resp);
+
```

Comment:
i dont think sending a 400 is here expected. This condition happens if there was no client id found for this app. So we should return an output like:

```
{status: "CLIENT_DOES_NOT_EXIST_ERROR"}
```

and for the case of when it does exist, we can return:
```
{status: "OK", ...rest of the json fields}
```

Reply:
Both with a status of 200

Reply:
done